File: snippets/node/handling_the_webhook_response_in_your_application.js
const express = require("express");
const bodyParser = require("body-parser");
// this is a stand-in for the code you'd use to write to your own database
const fakeDB = require("fakeDB");

const app = express();

app.use(bodyParser.json())
app.use(bodyParser.urlencoded({ extended: true }));

app.post("/", (req, res) => {
 const { type, data } = req.body;
 if (type === "subscribe") {
   fakeDB.subscribeUser(data);
 } else if (type === "unsubscribe") {
   fakeDB.unsubscribeUser(data.id);
 }
});

app.listen(port, () =>
 console.log(`Listening at http://localhost:3000`)
);

File: snippets/ruby/handling_the_webhook_response_in_your_application.rb
require 'sinatra'

configure do
  set :port, 3000
end


post '/' do
  type, data = params['type'], params['data']

  if type == 'unsubscribe'
    fake_db.unsubscribe_user(data['id'])
  elsif type == 'subscribe'
    fake_db.subscribe(data)
  end
end